### PR TITLE
Automated cherry pick of #274: central dashboard: parameterize userid-{header, prefix} Cherry pick of #274 on v0.6-branch. #274: central dashboard: parameterize userid-{header, prefix}

### PR DESCRIPTION
**Pull request by jlewi**
Related to kubeflow/kubeflow#3986

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/305)

### CODE REVIEW

**Comment by jlewi** [comment]
Resulting deployment looks good for IAP config

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    app: centraldashboard
    app.kubernetes.io/component: centraldashboard
    app.kubernetes.io/instance: centraldashboard
    app.kubernetes.io/managed-by: kfctl
    app.kubernetes.io/name: centraldashboard
    app.kubernetes.io/part-of: kubeflow
    app.kubernetes.io/version: v0.6
    kustomize.component: centraldashboard
  name: centraldashboard
  namespace: kubeflow
spec:
  replicas: 1
  selector:
    matchLabels:
      app: centraldashboard
      app.kubernetes.io/component: centraldashboard
      app.kubernetes.io/instance: centraldashboard
      app.kubernetes.io/managed-by: kfctl
      app.kubernetes.io/name: centraldashboard
      app.kubernetes.io/part-of: kubeflow
      app.kubernetes.io/version: v0.6
      kustomize.component: centraldashboard
  template:
    metadata:
      labels:
        app: centraldashboard
        app.kubernetes.io/component: centraldashboard
        app.kubernetes.io/instance: centraldashboard
        app.kubernetes.io/managed-by: kfctl
        app.kubernetes.io/name: centraldashboard
        app.kubernetes.io/part-of: kubeflow
        app.kubernetes.io/version: v0.6
        kustomize.component: centraldashboard
    spec:
      containers:
      - env:
        - name: USERID_HEADER
          value: ""
        - name: USERID_PREFIX
          value: ""
        - name: PROFILES_KFAM_SERVICE_HOST
          value: profiles-kfam.kubeflow
        image: gcr.io/kubeflow-images-public/centraldashboard:v20190822-v0.6.0-rc.0-68-g127e6006
        imagePullPolicy: IfNotPresent
        name: centraldashboard
        ports:
        - containerPort: 8082
          protocol: TCP
      serviceAccountName: centraldashboard

```

File: tests/centraldashboard-base_test.go [hunk 1]
```diff
@@ -76,8 +76,11 @@ spec:
           value: $(userid-header)
         - name: USERID_PREFIX
           value: $(userid-prefix)
+<<<<<<< HEAD
         - name: PROFILES_KFAM_SERVICE_HOST
           value: profiles-kfam.kubeflow
+=======
```
Comment (avdaredevil) [inline tests/centraldashboard-base_test.go]: @jlewi This seems to be a relic of a merge conflict that wasn't resolved and committed?

**Pull request merged**